plugins/bulk-import/src/components/AddRepositories/RepositoriesTableBody.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+
+import { makeStyles } from '@material-ui/core';
+import CircularProgress from '@mui/material/CircularProgress';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableRow from '@mui/material/TableRow';
+import { useFormikContext } from 'formik';
+
+import {
+  AddedRepositories,
+  AddRepositoriesFormValues,
+  AddRepositoryData,
+  RepositoryStatus,
+} from '../../types/types';
+import { OrganizationTableRow } from './OrganizationTableRow';
+import { RepositoriesColumnHeader } from './RepositoriesColumnHeader';
+import { RepositoryTableRow } from './RepositoryTableRow';
+
+const useStyles = makeStyles(theme => ({
+  empty: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+export const RepositoriesTableBody = ({
+  loading,
+  ariaLabel,
+  showOrganizations,
+  rows,
+  emptyRows,
+  onOrgRowSelected,
+  onClick,
+  selectedRepos,
+  isDrawer,
+}: {
+  loading: boolean;
+  ariaLabel: string;
+  emptyRows: number;
+  rows: AddRepositoryData[];
+  onOrgRowSelected: (org: AddRepositoryData) => void;
+  onClick: (_event: React.MouseEvent, repo: AddRepositoryData) => void;
+  selectedRepos: AddedRepositories;
+  isDrawer?: boolean;
+  showOrganizations?: boolean;
+}) => {
+  const classes = useStyles();
+  const { values } = useFormikContext<AddRepositoriesFormValues>();
+
+  const isSelected = (id: string) => {
+    return !!selectedRepos[id];
+  };
+
+  if (loading) {
+    return (
+      <tbody>
+        <tr>
+          <td colSpan={RepositoriesColumnHeader.length}>
+            <div data-testid={`${ariaLabel}-loading`} className={classes.empty}>
+              <CircularProgress />
+            </div>
+          </td>
+        </tr>
+      </tbody>
+    );
+  } else if (rows?.length > 0) {
+    return (
+      <TableBody>
+        {rows.map(row => {
```

Comment:
Is there a reason why this doesn't use the Table component from `@backstage/core-component`?

Reply:
Understand that we use MUI table here to customize the headers and custom selection checkboxes.

Unsure if there are alternatives with backstage-components here, but that's fine for now.